vllm/entrypoints/disaggregated/connector.py

```diff
@@ -0,0 +1,131 @@
+# SPDX-License-Identifier: Apache-2.0
+
+import uvicorn
+import uvloop
+
+from collections.abc import AsyncIterator
+from contextlib import asynccontextmanager
+from fastapi import FastAPI, Request
+from fastapi.responses import JSONResponse, StreamingResponse
+
+from vllm.entrypoints.disaggregated.engine import PDEngine
+from vllm.entrypoints.openai.serving_completion import OpenAIServingCompletion
+from vllm.entrypoints.openai.serving_models import (BaseModelPath,
+                                                    OpenAIServingModels)
+from vllm.entrypoints.openai.protocol import CompletionRequest
+from vllm.logger import init_logger
+from vllm.utils import FlexibleArgumentParser, set_ulimit, make_zmq_socket
+from vllm.entrypoints.openai.protocol import (
+    CompletionResponse, ErrorResponse)
+
+# Cannot use __name__ (https://github.com/vllm-project/vllm/pull/4765)
+logger = init_logger('vllm.entrypoints.disaggregated.api_server')
+
+app = FastAPI()
+
+@app.get("/v1/models")
+async def show_available_models(raw_request: Request):
+    handler: OpenAIServingModels = raw_request.app.state.openai_serving_models
+    models_ = await handler.show_available_models()
+    return JSONResponse(content=models_.model_dump())
+
+@app.post("/v1/completions")
+async def create_completion(request: CompletionRequest, raw_request: Request):
+    handler: OpenAIServingCompletion = raw_request.app.state.openai_serving_completion
+    generator = await handler.create_completion(request, raw_request)
+    if isinstance(generator, ErrorResponse):
+        return JSONResponse(content=generator.model_dump(),
+                            status_code=generator.code)
+    elif isinstance(generator, CompletionResponse):
+        return JSONResponse(content=generator.model_dump())
+
+    return StreamingResponse(content=generator, media_type="text/event-stream")
+
+@asynccontextmanager
+async def pd_engine_client_ctx_manager(
+    prefill_addr: str,
+    decode_addr: str,
+    connector_addr: str,
+    model_name: str
+) -> AsyncIterator[PDEngine]:
+    engine = PDEngine(prefill_addr, decode_addr, connector_addr, model_name)
```

Comment:
you can set up a side thread to run a busy loop, to receive register / unregister / heartbeat signals from entrypoint (or i would name it api servers) and engines.